### PR TITLE
Reset password

### PR DESCRIPTION
**Pull request by devtuananh**

### CODE REVIEW

File: app/controllers/password_resets_controller.rb [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+  def edit; end
+
+  def update
+    if params[:user][:password].empty?
+      @user.errors.add :password, t("password_resets.flash.password_not_empty")
+      render :edit
+    elsif @user.update_attributes user_params
+      log_in @user
+      @user.update_attribute :reset_digest, nil
```
Comment (mdmahmudurrahman) [inline app/controllers/password_resets_controller.rb]: Why update_attribute is used!

File: app/mailers/user_mailer.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+class UserMailer < ApplicationMailer
+
+  def account_activation(user)
+    @user = user
+    mail to: user.email, subject: "Account activation"
+  end
+
+  def password_reset(user)
```
Comment (mdmahmudurrahman) [inline app/mailers/user_mailer.rb]: Remove ()

File: app/mailers/user_mailer.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+class UserMailer < ApplicationMailer
+
+  def account_activation(user)
```
Comment (mdmahmudurrahman) [inline app/mailers/user_mailer.rb]: remove ()

File: app/mailers/user_mailer.rb [hunk 4]
```diff
@@ -0,0 +1,12 @@
+class UserMailer < ApplicationMailer
+
```
Comment (mdmahmudurrahman) [inline app/mailers/user_mailer.rb]: remove empty line

File: app/mailers/user_mailer.rb [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+  def account_activation(user)
+    @user = user
+    mail to: user.email, subject: "Account activation"
```
Comment (mdmahmudurrahman) [inline app/mailers/user_mailer.rb]: If you are using this method, please use I18n, else remove the method

File: app/models/user.rb [hunk 6]
```diff
@@ -35,6 +35,20 @@ def forget
     update_attributes remember_digest: User.digest(remember_token)
   end
 
+  def create_reset_digest
+    self.reset_token = User.new_token
+    update_attribute(:reset_digest,  User.digest(reset_token))
+    update_attribute(:reset_sent_at, Time.zone.now)
```
Comment (mdmahmudurrahman) [inline app/models/user.rb]: 1. You can combile line 40 and 41 into one. 
2. Use update_attributes

File: app/views/layouts/_header.html.erb [hunk 7]
```diff
@@ -65,6 +65,7 @@
                   <% end %>
                 </li>
               <% end %>
+
```
Comment (mdmahmudurrahman) [inline app/views/layouts/_header.html.erb]: Remove empty line

File: app/views/layouts/_header.html.erb [hunk 8]
```diff
@@ -65,6 +65,7 @@
...
               </ul>
```
Comment (mdmahmudurrahman) [inline app/views/layouts/_header.html.erb]: Indentation is not right.

**Comment by devtuananh** [comment]
ready

File: config/routes.rb [hunk 9]
```diff
@@ -1,9 +1,11 @@
 Rails.application.routes.draw do
+  get "static_pages/home"
```
Comment (mdmahmudurrahman) [inline config/routes.rb]: Already defined root_path with the same routing don't serve the purpose!

Comment (devtuananh) [reply config/routes.rb]: Yes

File: app/models/user.rb (hunk 6 above)
Comment (devtuananh) [reply app/models/user.rb]: Yes, I for got it. Sorry bro

File: app/views/layouts/_header.html.erb (hunk 8 above)
Comment (devtuananh) [reply app/views/layouts/_header.html.erb]: I can indent right <% end %> 2 space, really ? bro

File: app/controllers/password_resets_controller.rb (hunk 1 above)
Comment (devtuananh) [reply app/controllers/password_resets_controller.rb]: I edited to update_attributes

**Comment by devtuananh** [comment]
ready

File: app/views/layouts/_header.html.erb [hunk 10]
```diff
@@ -39,21 +39,30 @@
...
             </div>
             <div class="collapse navigation navbar-collapse navbar-ex1-collapse remove-space">
               <ul class="list-unstyled nav1 cl-effect-10">
                 <li><a data-hover="<%= t ".home" %>" class="active" href="#"><span><%= t ".home" %></span></a></li>
                 <li><a data-hover="<%= t ".about" %>" href="#"><span><%= t ".about" %></span></a></li>
                 <li><a data-hover="<%= t ".rooms" %>" href="#"><span><%= t ".rooms" %></span></a></li>
                 <li><a data-hover="<%= t ".contact_us" %>" href="#"><span><%= t ".contact_us" %></span></a></li>
-              <% if logged_in? %>
+                <% if logged_in? %>
+                <li>
```
Comment (mdmahmudurrahman) [inline app/views/layouts/_header.html.erb]: 2 space indentation

**Comment by devtuananh** [comment]
ready

**Comment by devtuananh** [comment]
ready

**Comment by devtuananh** [comment]
ready

**Comment by devtuananh** [comment]
ready

**Comment by devtuananh** [comment]
ready

**Comment by devtuananh** [comment]
ready

**Comment by devtuananh** [comment]
ready

**Pull request merged**